src/pubsub/ua_pubsub_networkmessage.c

```diff
@@ -1430,6 +1430,11 @@ UA_DataSetMessage_encodeBinary(const UA_DataSetMessage* src, UA_Byte **bufPos,
     UA_StatusCode rv = UA_DataSetMessageHeader_encodeBinary(&src->header, bufPos, bufEnd);
     UA_CHECK_STATUS(rv, return rv);
 
+    if(src->data.keyFrameData.fieldCount == 0) {
```

Comment:
What is the relation of **6.2.4.3 KeyFrameCount** and the `keyFrameData.fieldCount`?

Reply:
There is no relation.
KeyFrameCount is a parameter of the Writer, which controls the frequency of Key frames in the Key/Delta frame sequence.
keyFrameData.fieldCount is a field in the message 7.2.2.5.5 Data Key Frame DataSetMessage